lib/matplotlib/_pylab_helpers.py

```diff
@@ -47,6 +59,50 @@ def get_fig_manager(cls, num):
         return manager
 
     @classmethod
+    def show_all(cls, block=None):
+        """
+        Show all figures.  If *block* is not None, then
+        it is a boolean that overrides all other factors
+        determining whether show blocks by calling mainloop().
+        The other factors are:
+        it does not block if run inside ipython's "%pylab" mode
+        it does not block in interactive mode.
+        """
+
+        managers = cls.get_all_fig_managers()
+        if not managers:
+            return
+
+        for manager in managers:
+            manager.show()
+
+        if block is not None:
+            if block:
```

Comment:
To be clear: I'm not saying there is anything wrong with the code; on the contrary, I am replying to @weathergod, to say there is a good reason for it being exactly the way it is, at least in its original form and location. I think he was questioning the reason for having a check for None followed by a check for True.